src/richie/apps/courses/factories.py

```diff
@@ -141,17 +143,121 @@ def extended_object(self):
         """
         Automatically create a related page with the random title
         """
-        return create_page(self.title, Course.TEMPLATE_DETAIL, settings.LANGUAGE_CODE)
+        return create_page(
+            self.title,
+            Course.TEMPLATE_DETAIL,
+            settings.LANGUAGE_CODE,
+            parent=self.parent,
+        )
 
     @factory.post_generation
-    # pylint: disable=unused-argument, attribute-defined-outside-init, no-member
+    # pylint: disable=unused-argument
+    def fill_teaser(self, create, extracted, **kwargs):
+        """
+        Add a video plugin for teaser with a random url
+        """
+        if create and extracted:
+            language = settings.LANGUAGE_CODE
+
+            placeholder = self.extended_object.placeholders.get(slot="course_teaser")
+
+            label, url = random.choice(VIDEO_SAMPLE_LINKS)
+
+            add_plugin(
+                language=language,
+                placeholder=placeholder,
+                plugin_type="VideoPlayerPlugin",
+                label=label,
+                embed_link=url,
+            )
+
+    @factory.post_generation
+    # pylint: disable=unused-argument
+    def fill_syllabus(self, create, extracted, **kwargs):
+        """
+        Add a text plugin for syllabus with a long random text
+        """
+        if create and extracted:
+            create_text_plugin(self.extended_object, "course_syllabus", nb_paragraphs=1)
+
+    @factory.post_generation
+    # pylint: disable=unused-argument
+    def fill_format(self, create, extracted, **kwargs):
+        """
+        Add a text plugin for course format with a long random text
+        """
+        if create and extracted:
+            create_text_plugin(self.extended_object, "course_format", nb_paragraphs=1)
+
+    @factory.post_generation
+    # pylint: disable=unused-argument
+    def fill_prerequisites(self, create, extracted, **kwargs):
+        """
+        Add a text plugin for pre requisites with a long random text
+        """
+        if create and extracted:
+            create_text_plugin(
+                self.extended_object, "course_prerequisites", nb_paragraphs=1
+            )
+
+    @factory.post_generation
+    # pylint: disable=unused-argument
+    def fill_plan(self, create, extracted, **kwargs):
```

Comment:
Sorry I just thougt of something. Since this is so repetitive maybe we could have only one method for all text placeholders and pass the list of placeholders we want : 

CourseFactory(fill_paragraphs=["plan", "teaser", ...])